apm-agent-plugins/apm-jmx-plugin/src/test/java/co/elastic/apm/agent/jmx/JmxMetricTrackerTest.java

```diff
@@ -111,13 +116,54 @@ void testRemoveMetric() throws Exception {
         }
     }
 
+    @Test
+    void testMBeanAddedLater() throws Exception {
+        setConfig(JmxMetric.valueOf("object_name[foo:type=Bar] attribute[Baz]"));
+        assertThat(jmxMetricTracker.getRegisteredListeners()).hasSize(1);
+        ObjectName objectName = new ObjectName("foo:type=Bar");
+        ManagementFactory.getPlatformMBeanServer().registerMBean(new TestMetric(), objectName);
+        try {
+            assertThat(jmxMetricTracker.getRegisteredListeners()).isEmpty();
+            assertThat(metricRegistry.getGauge("jvm.jmx.Baz", Labels.Mutable.of("type", "Bar"))).isEqualTo(42);
+        } finally {
+            ManagementFactory.getPlatformMBeanServer().unregisterMBean(objectName);
+        }
```

Comment:
we probably miss an assertion on the number of listeners, which should be equal to zero when the bean is available. Also, what happens if a bean is registered only from time to time (thus multiple register/unregister cycles) ?

Reply:
> we probably miss an assertion on the number of listeners, which should be equal to zero when the bean is available.

Isn't that done in line 126?

> Also, what happens if a bean is registered only from time to time

not quite sure what you mean with that